From 49cb7f163fc6beaf2a07b12d6bcb0786989ecee9 Mon Sep 17 00:00:00 2001
From: squalus <squalus@squalus.net>
Date: Mon, 16 May 2022 00:27:59 -0700
Subject: [PATCH] buildMozillaMach: fix builds with crash reporting disabled

- only try to build and copy crash reporting symbols when crash
  reporting is enabled
- fixes the librewolf build
---
 .../networking/browsers/firefox/common.nix           | 12 +++++-------
 1 file changed, 5 insertions(+), 7 deletions(-)

diff --git a/pkgs/applications/networking/browsers/firefox/common.nix b/pkgs/applications/networking/browsers/firefox/common.nix
index 25eb34c95e8268..4416c2a36376d4 100644
--- a/pkgs/applications/networking/browsers/firefox/common.nix
+++ b/pkgs/applications/networking/browsers/firefox/common.nix
@@ -169,10 +169,8 @@ buildStdenv.mkDerivation ({
 
   inherit src unpackPhase meta;
 
-  outputs = [
-    "out"
-    "symbols"
-  ];
+  outputs = [ "out" ]
+  ++ lib.optionals crashreporterSupport [ "symbols" ];
 
   # Add another configure-build-profiling run before the final configure phase if we build with pgo
   preConfigurePhases = lib.optionals pgoSupport [
@@ -202,7 +200,6 @@ buildStdenv.mkDerivation ({
   nativeBuildInputs = [
     autoconf
     cargo
-    dump_syms
     llvmPackages.llvm # llvm-objdump
     makeWrapper
     nodejs
@@ -217,6 +214,7 @@ buildStdenv.mkDerivation ({
     wrapGAppsHook
   ]
   ++ lib.optionals pgoSupport [ xvfb-run ]
+  ++ lib.optionals crashreporterSupport [ dump_syms ]
   ++ extraNativeBuildInputs;
 
   setOutputFlags = false; # `./mach configure` doesn't understand `--*dir=` flags.
@@ -421,11 +419,11 @@ buildStdenv.mkDerivation ({
 
   # Generate build symbols once after the final build
   # https://firefox-source-docs.mozilla.org/crash-reporting/uploading_symbol.html
-  preInstall = ''
+  preInstall = lib.optionalString crashreporterSupport ''
     ./mach buildsymbols
     mkdir -p $symbols/
     cp mozobj/dist/*.crashreporter-symbols.zip $symbols/
-
+  '' + ''
     cd mozobj
   '';
